src/main/java/com/gluonhq/substrate/target/AbstractTargetConfiguration.java

```diff
@@ -263,19 +265,29 @@ public boolean compileAdditionalSources()
     }
 
     @Override
-    public InputStream run(Path appPath, String appName) throws IOException {
-        Process runProcess = startAppProcess(appPath,appName);
-        return runProcess.getInputStream();
+    public String run(Path appPath, String appName) throws IOException, InterruptedException {
+        Path app = Objects.requireNonNull(appPath, "Application path can't be null")
+                .resolve(Objects.requireNonNull(appName, "Application name can't be null"));
+        if (!Files.exists(app)) {
+            throw new IOException("Application not found at path " + app.toString());
+        }
+        ProcessRunner runner = new ProcessRunner(app.toString());
```

Comment:
Can you file an issue for creating docs + tests for ProcessRunner? From the method names and signature, it is not clear what they are doing. You currently need to investigate the code to find out, and those methods can have serious impacts on performance, hence developers want to know what they do.

Reply:
Done